pkg/occlient/occlient.go

```diff
@@ -843,12 +815,66 @@ func (c *Client) BootstrapSupervisoredS2I(name string, builderImage string, labe
 	return nil
 }
 
+// AddBootstrapInitContainer adds the bootstrap init container to the deployment config
+// dc is the deployment config to be updated
+// dcName is the name of the deployment config
+func (c *Client) AddBootstrapInitContainer(dc *appsv1.DeploymentConfig, dcName string) {
+	appRootVolumeName := fmt.Sprintf("%s-s2idata", dcName)
+	dc.Spec.Template.Spec.InitContainers = append(dc.Spec.Template.Spec.InitContainers,
+		corev1.Container{
+			Name:  "copy-files-to-volume",
+			Image: dc.Spec.Template.Spec.Containers[0].Image,
+			Command: []string{
+				"copy-files-to-volume",
+				"/opt/app-root",
+				"/mnt/app-root"},
+			VolumeMounts: []corev1.VolumeMount{
+				{
+					// TODO:
+					Name:      appRootVolumeName,
+					MountPath: "/mnt",
+				},
+			},
+		})
+}
+
+// AddBootstrapVolume adds the bootstrap volume to the deployment config
+// dc is the deployment config to be updated
+// dcName is the name of the deployment config
+func (c *Client) AddBootstrapVolume(dc *appsv1.DeploymentConfig, dcName string) {
+	appRootVolumeName := fmt.Sprintf("%s-s2idata", dcName)
+	dc.Spec.Template.Spec.Volumes = append(dc.Spec.Template.Spec.Volumes, corev1.Volume{
+		Name: appRootVolumeName,
+		VolumeSource: corev1.VolumeSource{
+			PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+				ClaimName: appRootVolumeName,
+			},
+		},
+	})
+}
+
+// AddBootstrapVolumeMount mounts the bootstrap volume to the deployment config
+// dc is the deployment config to be updated
+// dcName is the name of the deployment config
+func (c *Client) AddBootstrapVolumeMount(dc *appsv1.DeploymentConfig, dcName string) {
+	appRootVolumeName := fmt.Sprintf("%s-s2idata", dcName)
+	for i := range dc.Spec.Template.Spec.Containers {
+		dc.Spec.Template.Spec.Containers[i].VolumeMounts = append(dc.Spec.Template.Spec.Containers[i].VolumeMounts, corev1.VolumeMount{
+			Name:      appRootVolumeName,
+			MountPath: "/opt/app-root",
+			SubPath:   "app-root",
+		})
+	}
+}
+
 // UpdateBuildConfig updates the BuildConfig file
 // buildConfigName is the name of the BuildConfig file to be updated
 // projectName is the name of the project
 // gitUrl equals to the git URL of the source and is equals to "" if the source is of type dir or binary
 // annotations contains the annotations for the BuildConfig file
 func (c *Client) UpdateBuildConfig(buildConfigName string, projectName string, gitUrl string, annotations map[string]string) error {
+	const bootstrapperURI = "https://github.com/kadel/bootstrap-supervisored-s2i"
```

Comment:
this is duplicate, it is already defined in Boostrap function

Reply:
Should I make it a global variable?